### PR TITLE
feat: Cost Center selection for checkout

### PR DESCRIPTION
**Pull request by cbratsch**
## PR Type

[x] Feature

## What Is the Current Behavior?
It is currently not possible to successfully enter a B2B checkout process if the logged in user is assigned to any cost center. For this a cost center selection is missing at the shopping cart page.

## What Is the New Behavior?
Business Users can select a cost center in the checkout process, when assigned to one or multiple cost centers.

## Does this PR Introduce a Breaking Change?

[x] No

## Other Information

This feature does not include Cost Center Management functionalities.

[AB#65623](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/65623)

### CODE REVIEW

File: src/app/shared/components/checkout/cost-center-selection/cost-center-selection.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<ng-container *ngIf="showSelection">
```
Comment (SGrueber) [inline src/app/shared/components/checkout/cost-center-selection/cost-center-selection.component.html:1]: (costCenterOptions$ | async)?.length && isBusinessCustumer | async

something like this

**Review by SGrueber — CHANGES_REQUESTED** [review]

**Review by M-Behr — CHANGES_REQUESTED** [review]
Hi, it works really well. Here are some issues:

- [ ] Cost Center ID and name should be displayed in the buyer box at the end of checkout.
- [x] You select a cost center and go to checkout. If you click "Return to Cart" or you come back to the cart page somehow else, the cost center selection is empty. The selected cost center should be displayed. - this needs the appropriate fix of the ICM cost center REST interface (latest LTS patch)

**Review by iwiederhold — APPROVED** [review]
Looks fine!

**Review by M-Behr — COMMENTED** [review]
Great job! 👍

**Review by M-Behr — APPROVED** [review]

**Pull request merged**